lib/client/api.go

```diff
@@ -714,6 +714,10 @@ func RetryWithRelogin(ctx context.Context, tc *TeleportClient, fn func() error)
 		return nil
 	}
 
+	if utils.IsPredicateError(err) {
+		return trace.Wrap(utils.PredicateError{Err: err})
```

Comment:
If it's already a predicate error, why do you have to wrap it in another predicate error?

Reply:
the [IsPredicateError](https://github.com/gravitational/teleport/blob/master/lib/utils/cli.go#L500) looks for keyword in the error to determine if its predicate, then we use a wrapper to add a link to the predicate documentation: 

so if this is the error that returned:
```
failed to parse predicate expression: identifier "la" is not defined
```

it will get formatted to this:
```
ERROR: failed to parse predicate expression: identifier "la" is not defined
Check syntax at https://goteleport.com/docs/setup/reference/predicate-language/#resource-filtering
```

Reply:
It feels to me that `RetryWithRelogin` shouldn't be concerned with doing special stuff with errors that have nothing to do with auth. 🤔

Is there a single place which could wrap the predicate error before `RetryWithRelogin` catches it? Then we could change `IsErrorResolvableWithRelogin` to return `false` for predicate errors and that would be it.

If that's not possible, then perhaps another solution would be to make `IsErrorResolvableWithRelogin` return `false` for predicate errors and then wrap them in specific callsites, as it was done before this PR.